### PR TITLE
ente-auth: fix build failure because of wrong icon path

### PR DESCRIPTION
**Pull request by MrSom3body**
Fix [build failure](https://gist.github.com/MrSom3body/76f799b736df35d285149d048192823b) for ente-auth.

ZHF: #352882

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by schnow265 — APPROVED** [review]

**Review by gepbird — CHANGES_REQUESTED** [review]
Thanks for your contribution!

@MrSom3body can you share more details about what are you trying to fix? This package builds for me on master, but I get this error with your change:

```
install: cannot stat '/nix/store/dxgqlaxsjydgxshh29f0qsz7jszlkdgs-ente-auth-4.0.2/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
```

To test the build yourself, you can run `nix-build -A ente-auth` from the root of your fork of nixpkgs.

File: pkgs/by-name/en/ente-auth/package.nix [hunk 1]
```diff
@@ -78,7 +78,7 @@ flutter324.buildFlutterApplication rec {
   ];
 
   postInstall = ''
-    FAV=$out/app/data/flutter_assets/assets/icons/auth-icon.png
+    FAV=$out/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png
     ICO=$out/share/icons
 
     install -D $FAV $ICO/ente-auth.png
```
Comment (Aleksanaa) [inline pkgs/by-name/en/ente-auth/package.nix:84]:
```suggestion
    install -D $out/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png $out/share/pixmaps/ente-auth.png
```

Delete the below lines. These should be useless

Comment (gepbird) [reply pkgs/by-name/en/ente-auth/package.nix:84]: `ICO` and `FAV` are also used in a for loop right below, I think its cleaner to keep those variables.

Comment (Aleksanaa) [reply pkgs/by-name/en/ente-auth/package.nix:84]: I'm saying the following should be removed. share/pixmaps is stipulated by the freedesktop specification, and this path does not need to specify the size. I don't think zooming with imagemagick is a good way.

Also remove `imagemagick` from `nativeBuildInputs` and function formal.

Comment (gepbird) [reply pkgs/by-name/en/ente-auth/package.nix:84]: Thanks for explaining this! Even though `pixmaps` is deprecated, it seems like better choice than creating images with `imagemagick`.

**Comment by MrSom3body** [comment]
> Thanks for your contribution!
> 
> @MrSom3body can you share more details about what are you trying to fix? This package builds for me on master, but I get this error with your change:
> 
> ```
> install: cannot stat '/nix/store/dxgqlaxsjydgxshh29f0qsz7jszlkdgs-ente-auth-4.0.2/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
> ```
> 
> To test the build yourself, you can run `nix-build -A ente-auth` from the root of your fork of nixpkgs.

Huh, that's weird, I get the same error message if I do not have the change. Here's my [log](https://gist.github.com/MrSom3body/76f799b736df35d285149d048192823b).

**Comment by niklaskorz** [comment]
> > Thanks for your contribution!
> > @MrSom3body can you share more details about what are you trying to fix? This package builds for me on master, but I get this error with your change:
> > ```
> > install: cannot stat '/nix/store/dxgqlaxsjydgxshh29f0qsz7jszlkdgs-ente-auth-4.0.2/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To test the build yourself, you can run `nix-build -A ente-auth` from the root of your fork of nixpkgs.
> 
> Huh, that's weird, I get the same error message if I do not have the change. Here's my [log](https://gist.github.com/MrSom3body/76f799b736df35d285149d048192823b).
> 
> Going into the directory that it build, I can see the following directory structure:

Are you also using Lix, like the original reporter of the build failure?

**Comment by MrSom3body** [comment]
Yes, I am. So it seems to be a problem with lix and not with the package itself. Should I close the pull request then?

**Comment by niklaskorz** [comment]
> > install: cannot stat '/nix/store/q0ckhxy060rb82714z391xl0gfisqrsp-ente-auth-4.0.2/app/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
> 
> I get this error also with `nix` on both `nixos-unstable-small` and `master`. Tested with:
> 
>     * nix-2.18.8
> 
>     * nix-2.24.9
> 
>     * lix-2.90.0
> 
>     * lix-2.91.1
> 
> 
> (I am also a `lix` user but tested and build in `nix-shell -p nixVersions.nix_2_18` etc)

Spawning a shell is insufficient. The Nix daemon is unaffected by this.

**Comment by niklaskorz** [comment]
> > > > install: cannot stat '/nix/store/q0ckhxy060rb82714z391xl0gfisqrsp-ente-auth-4.0.2/app/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
> > > 
> > > 
> > > I get this error also with `nix` on both `nixos-unstable-small` and `master`. Tested with:
> > > ```
> > > * nix-2.18.8
> > > 
> > > * nix-2.24.9
> > > 
> > > * lix-2.90.0
> > > 
> > > * lix-2.91.1
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > (I am also a `lix` user but tested and build in `nix-shell -p nixVersions.nix_2_18` etc)
> > 
> > 
> > Spawning a shell is insufficient. The Nix daemon is unaffected by this.
> 
> Sorry, didn't think. I tested an fedora, my daemon is from `nix-2.21.2` 🤷‍♂️

Yeah can confirm for 2.18 now (back at my NixOS machine). And nice find. I just wonder why @chewblacka said the build worked for them without this fix.

File: pkgs/by-name/en/ente-auth/package.nix (hunk 1 above)
Comment (niklaskorz) [reply pkgs/by-name/en/ente-auth/package.nix:84]: Let's move this to a follow-up PR as it's irrelevant to this fix, so we can unbreak the build first.

**Review by niklaskorz — COMMENTED** [review]
Building this PR fails for me:

```
       > install: cannot stat '/nix/store/dxgqlaxsjydgxshh29f0qsz7jszlkdgs-ente-auth-4.0.2/app/ente-auth/data/flutter_assets/assets/icons/auth-icon.png': No such file or directory
       For full logs, run 'nix log /nix/store/xsw15jfkniv34zbvlqc8s0d0hcg7ldml-ente-auth-4.0.2.drv'.
```

**Review by zi3m5f — APPROVED** [review]
This fixes the error introduced by #345404 for me :+1:

**Comment by Aleksanaa** [comment]
This is not needed, because ofborg will rebase itself to master before testing, so does nixpkgs-review

**Comment by niklaskorz** [comment]
> This is not needed, because ofborg will rebase itself to master before testing, so does nixpkgs-review

Yeah just wanted to clarify why it did not build for me at first :)

**Review by niklaskorz — APPROVED** [review]

**Comment by niklaskorz** [comment]
@MrSom3body can you add this text to the PR description, as this PR is part of ZHF:

> ZHF: #352882

**Comment by MrSom3body** [comment]
Will do when I'm back home

**Comment by gepbird** [comment]
I tested it with `nix-build` and `nom build` in my local nixpkgs on NixOS, I'm also using Lix. Works an master, fails on this PR.

**Review by zi3m5f — APPROVED** [review]

**Review by gepbird — COMMENTED** [review]
Builds on x86_64-linux, thanks!

**Pull request merged**